### ISSUE TITLE
formatter: Diff with Prettier on trailing spaces in alignable comment

### ISSUE THREAD

**Issue by leaysgur** [oxc-project/oxc#17786]
### Input

```tsx
/* See -> 
   this  ^
   trailing space is trimmed 
*/

```

### Config

```jsonc
{}
```

### Oxfmt output

Oxfmt version: `dev`
```tsx
/* See ->
   this  ^
   trailing space is trimmed 
*/

```

### Prettier output

Prettier version: `3.7.4`
```tsx
/* See -> 
   this  ^
   trailing space is trimmed 
*/

```

### Additional notes

Please refer these actual files in `vercel/next.js` repo.

- examples/with-http2/next.config.js
- packages/eslint-plugin-next/src/utils/node-attributes.ts
- packages/next/src/next-devtools/dev-overlay/shared.ts
- test/production/app-dir/actions-tree-shaking/_testing/utils.ts

**Comment by Dunqing** [comment]
I have been aware of this difference for a while. I think we don't need to align this, as this is a pointless space, and most people will remove it. Also, trailing spaces are not easy to spot, so I think we made a good change for this.

**Comment by leaysgur** [comment]
While checking Prettier's issues, I found something interesting. 😶‍🌫️

https://github.com/prettier/prettier/issues/6793

In JSDoc (alignable one), trailing spaces have meaning when rendered as Markdown...!

||oxfmt|prettier|people|
|----|-----|--------|------|
|Block comment|trim|keep|???|
|JSDoc comment|trim|trim|keep|

**Comment by Dunqing** [comment]
https://github.com/prettier/prettier/issues/6793#issuecomment-1816740289 Looks good

**Comment by leaysgur** [comment]
- https://github.com/prettier/prettier/pull/18594
- https://github.com/prettier/prettier/issues/18602

**Comment by Dunqing** [comment]
Does the Next.js case have one space or two?

**Comment by leaysgur** [comment]
All Next.js cases are using 1 extra space, and not JSDoc, normal block comment.
